handlers.go

```diff
@@ -58,21 +58,27 @@ func BeginAuth(w http.ResponseWriter, r *http.Request) {
     }
 }
 
-func getSingleSecQues(roll string) string {
+func getSingleSecQues(roll string) {
+
     v := url.Values{}
     v.Set("user_id", roll)
     resp, _ := http.PostForm(ERP_SECRET_QUES_URL, v)
     body, _ := ioutil.ReadAll(resp.Body)
     return string(body)
+
 }
 
 func getSecurityQuestions(roll string) []string {
     allSecQues := []string{}
+    data := make(chan string)
 
-    // Perform upto 30 tries to get the 3 unique secret questions from ERP
-    for i := 1; i < 30; i++ {
-        secQues := getSingleSecQues(roll)
-        log.Printf("Run %d, Got %s", i, secQues)
+    for i := 1; i <= 30; i++ { // Perform upto 30 tries to get the 3 unique secret questions from ERP
+        go func() { data <- getSingleSecQues(roll) }()
+    }
+
+    for i := 1; i <= 30; i++ {
+        secQues := <-data
+        log.Printf("Run %d, Got %s\n", i, secQues)
         alreadyFound := false
         for _, q := range allSecQues {
```

Comment:
Perhaps, if it isn't too much work, you could change this to use a `map[string]bool`? This has been irking me for a while. Feel free to not do it in this PR.

Reply:
Oh. Ok. Sure!  Then, should this function return the map or the list of strings? The only place the function is called is here :
```
secQues := getSecurityQuestions(result.Roll)
for _, ques := range secQues {
            if strings.Contains(ques, result.Verifier) {
                verified = true
                break
            }
        }
```
The only modification would be then ->  `for ques,_ := range secQues ...`
What do you think? @icyflame

Reply:
No, no. I meant use a map instead of iterating through the whole list. The function should return a list.

Reply:
I have pushed a commit. Is that what what you are talking about?